src/client/renderingengine.h

```diff
@@ -137,9 +137,10 @@ class RenderingEngine
 		return m_device->getGUIEnvironment();
 	}
 
+	// Note that percent is ignored in case indefinite is true
 	void draw_load_screen(const std::wstring &text,
 			gui::IGUIEnvironment *guienv, ITextureSource *tsrc,
-			float dtime = 0, int percent = 0, bool sky = true);
+			float dtime = 0, int percent = 0, bool sky = true, bool indefinite = false);
```

Comment:
Is there any case where `percent == 0 && !indefinite`?

Reply:
alternatively overload `percent` so that -1 means indefinite